src/content/tools/doc-comment-references.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Documentation comment references
+short-title: Comment references
+description: Learn about doc comment references and their syntax.
+---
+
+Doc comments can contain references to various code elements. A library member
+or class member, etc. can be referenced by wrapping its name in square brackets
+(`[...]`) in a doc comment (a comment starting with `///`). Examples:
+
+```dart
+/// Returns a [String].
+String f() => 'Hello';
+
+/// Wraps [object] with [Future.value].
+Future<T> g<T>(T object) => Future.value(object);
+```
+
+These doc comments contain references to the `String` class, the `object`
+parameter, and the `Future.value` constructor.
+
+## Features of references
+
+There are several benefits to referring to a code element in a doc comment as a
+doc comment reference:
+
+### Editor features
+
+Doc comment references enable several IDE features. An element's name can be
+code-completed within square brackets. When an element is renamed via an IDE
+command, it can rewrite uses of that element, including references in doc
+comments. When an IDE lists all "references" to an element, it can include
+references in doc comments. An IDE can also provide Go-to-definition support at
+the location of a doc comment reference.
```

Comment:
Optionally consider making this a list, so readers can quickly see the features supported if they know them, and choose to learn more. For example:

```
Doc comment references enable several IDE features:

- **Code completion**
  An element's name can be code-completed within square brackets.
- **Rename refactoring**
  When an element is renamed via an IDE command, it can
  rewrite uses of that element, including references in doc comments.
```

Reply:
Great idea, done.